### ISSUE TITLE
spindle/cutter touching copper causes hard limit trigger

### ISSUE THREAD

**Issue by brentdbell** [grbl/grbl#1416]
I am trying to debug a problem with a new system setup.  My machine seems to work fine unless I try to make a PCB - I am trying to router a PCB, but I found that the system was hard resetting shortly after starting to cut the copper.  I took a single sided copper clad board and put the copper side down - and the system works just fine cutting the FR4 side, but as soon as I try cutting copper I get hard limits triggering.  If I raise up the z-level so that the cutter is a few mm away from the copper - it goes through the full g-code without a limit being triggered.

The copper clad board is 2-way taped to a piece of FR4 that is 2-way taped to another piece of FR4 - so it is definitely not touching anything else...

I am using a Chinese spindle and VFD - I have both the ground wire (pin 4) from the spindle and the shield from the spindle cable connected to PE on my inverter.

All of my cables are shielded with the shield connected to ground/earth on the CNC power supply. I have both de-coupling caps and pull-up resistors on the limit switches - what's wrong?  Any help is appreciated!

Update: I have added opto-couplers for the limit switches - still having problems.  It's not just when the spindle/cutter touches the copper, although sometimes it takes that to trigger the Alarm - it will also sometimes trigger the Alarm as soon as I turn up the speed of the motor.  If the VFD output is at 0 everything is fine, but as soon as I turn it on at all I can get an Alarm no matter where the Spindle/cutter is.  If I home the machine, move it to position, then disable the Hard limits I can router a PCB.  I believe I could also work with wood, fr4, etc - with the hard limits enabled...but not when working with metal. Anyone else see anything like this?

**Comment by Rob2029** [comment]
That's weird, if you can, check the voltage drop if there is, between the
spindle and ground when it is touching the copper, some odd parasite
inductance must be there, but I have no clue

**Comment by brentdbell** [comment]
I ended up solving the hard limit problem by opening up the end of the spindle and attaching the ground wire which was missing.  I had already opened the connector and connected it there, but found when I opened the spindle it was not connected inside.  I actually bypassed the opto-couplers and tested it - now no false hard limit triggering at all.

I am now trying to figure out how to control the spindle speed...but that is really beyond the scope of this forum so will post elsewhere...this post was also...but hopefully will help someone with similar problems.
